src/test/scala/mesosphere/marathon/test/GroupCreation.scala

```diff
@@ -3,15 +3,24 @@ package test
 
 import mesosphere.marathon.core.pod.PodDefinition
 import mesosphere.marathon.state._
+import com.wix.accord
 
 trait GroupCreation {
   def createRootGroup(
     apps: Map[AppDefinition.AppKey, AppDefinition] = Group.defaultApps,
     pods: Map[PathId, PodDefinition] = Group.defaultPods,
     groups: Set[Group] = Set.empty,
     dependencies: Set[PathId] = Group.defaultDependencies,
-    version: Timestamp = Group.defaultVersion): RootGroup = {
-    RootGroup(apps, pods, groups.map(group => group.id -> group)(collection.breakOut), dependencies, version)
+    version: Timestamp = Group.defaultVersion,
+    validate: Boolean = true): RootGroup = {
+    val group = RootGroup(apps, pods, groups.map(group => group.id -> group)(collection.breakOut), dependencies, version)
+
+    if (validate) {
+      val validation = accord.validate(group)(RootGroup.rootGroupValidator(Set("secrets")))
+      assert(validation.isSuccess, s"Provided test root group was not valid: ${validation.toString}")
```

Comment:
This is the important change. Everything else was introduced to make the test apps and groups valid.